superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx

```diff
@@ -0,0 +1,1307 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { FunctionComponent, useState, useEffect } from 'react';
+import { styled, t, SupersetClient } from '@superset-ui/core';
+import rison from 'rison';
+import { useSingleViewResource } from 'src/views/CRUD/hooks';
+
+import Icon from 'src/components/Icon';
+import Modal from 'src/common/components/Modal';
+import { Switch } from 'src/common/components/Switch';
+import { Select } from 'src/common/components/Select';
+import { Radio } from 'src/common/components/Radio';
+import { AsyncSelect } from 'src/components/Select';
+import withToasts from 'src/messageToasts/enhancers/withToasts';
+
+import Owner from 'src/types/Owner';
+import { AlertObject, Operator, Recipient, MetaObject } from './types';
+
+type SelectValue = {
+  value: string;
+  label: string;
+};
+
+interface AlertReportModalProps {
+  addDangerToast: (msg: string) => void;
+  alert?: AlertObject | null;
+  isReport?: boolean;
+  onAdd?: (alert?: AlertObject) => void;
+  onHide: () => void;
+  show: boolean;
+}
+
+const NOTIFICATION_METHODS: NotificationMethod[] = ['Email', 'Slack'];
+
+const CONDITIONS = [
+  {
+    label: '< (Smaller than)',
```

Comment:
we should probably wrap these in `t()` for translation

Reply:
opting to add the `t()` wrap in the Option component render below!

Reply:
Ah, that won't work. Translations only work with string literals that get extracted from source code into message catalogs which are then translated. The person translating the messages wouldn't know what they're translating if the message could be any string